purchase_discount/models/product_supplierinfo.py

```diff
@@ -27,9 +27,19 @@ def _get_po_to_supplierinfo_synced_fields(self):
     def create(self, vals_list):
         """Insert discount (or others) from context from purchase.order's
         _add_supplier_to_product method"""
-        for vals in vals_list:
-            product_tmpl_id = vals.get("product_tmpl_id")
-            po_line_map = self.env.context.get("po_line_map", {})
+        po_line_map = self.env.context.get("po_line_map")
+        if not po_line_map:
```

Comment:
I don't understand what this exit condition does.

Reply:
Well, I was in my way to explain how this is a dict injected in context with the only purpose of passing the discount values on creation when I realized that in this version there's a hook method that makes this engineering unnecessary